### PR TITLE
Gradle change in 2.2 and sdk update

### PR DESCRIPTION
**Pull request by Sadmansamee**
There is a change in Gradle 2.2
https://code.google.com/p/android/issues/detail?id=219732

### CODE REVIEW

**Comment by Sadmansamee** [comment]
I signed it!

**Comment by Sadmansamee** [comment]
I signed it!

File: WORKSPACE [hunk 1]
```diff
@@ -3,16 +3,16 @@ workspace(name = "org_tensorflow")
 # Uncomment and update the paths in these entries to build the Android demo.
 #android_sdk_repository(
 #    name = "androidsdk",
-#    api_level = 23,
-#    build_tools_version = "23.0.1",
+#    api_level = 25,
+#    build_tools_version = "24.0.2",
```
Comment (andrewharp) [inline WORKSPACE]: Is there a reason the api_level and build tools version shouldn't match?

File: WORKSPACE [hunk 2]
```diff
@@ -3,16 +3,16 @@ workspace(name = "org_tensorflow")
...
 #    # Replace with path to Android SDK on your system
 #    path = "<PATH_TO_SDK>",
 #)
 #
 #android_ndk_repository(
 #    name="androidndk",
 #    path="<PATH_TO_NDK>",
-#    api_level=21)
+#    api_level=25)
```
Comment (andrewharp) [inline WORKSPACE]: Is there a reason to update the native API level?

File: tensorflow/examples/android/build.gradle [hunk 3]
```diff
@@ -69,4 +69,8 @@ task copyNativeLibs(type: Copy) {
 
 copyNativeLibs.dependsOn buildNative
 assemble.dependsOn copyNativeLibs
-assembleDebug.dependsOn copyNativeLibs
+tasks.whenTaskAdded { task ->
```
Comment (andrewharp) [inline tensorflow/examples/android/build.gradle]: Can you add a comment to explain the reasoning behind this change please? I vaguely remember having to previously work around a bug in Gradle here -- just want to make sure this change will work under all circumstances going forward.

**Comment by andrewharp** [comment]
You'll need to update [tensorflow/tools/ci_build/Dockerfile.android](https://github.com/tensorflow/tensorflow/blob/754048a0453a04a761e112ae5d99c149eb9910dd/tensorflow/tools/ci_build/Dockerfile.android) as well so it matches the new defaults you've selected, and then we can test with Jenkins.

[tensorflow/tools/ci_build/builds/android.sh](https://github.com/tensorflow/tensorflow/blob/d44d271c9da4d244ce4b2ffaf808adbe4cff759d/tensorflow/tools/ci_build/builds/android.sh) pulls from it when auto-appending to the WORKSPACE file.

**Comment by danmane** [comment]
@Sadmansamee ping :)

**Comment by martinwicke** [comment]
Friendly Ping!

I'll close this as stale unless I hear from @Sadmansamee.

**Comment by drpngx** [comment]
Closing due to inactivity. Feel free to reopen once ready.

**Pull request closed without merging**